### PR TITLE
feat: build umd lib

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
No way to build a UMB lib

**What is the chosen solution to this problem?**
Use webpack to build the UMD via 
```
talend-scripts build:lib --env.umd=MyUMDName
talend-scripts build:lib --env.umd=MyUMDName --env.optimize
```

This example exposes MyUMDName module in the files
```
MyUMDName.js
MyUMDName.min.js
```

Limitations: for now it only support js/ts/tsx and svg files.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/config-react-webpack/config/utils/ts-config-exists.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const fs = require('fs');
+const path = require('path');
+
+module.exports = function tsConfigExists() {
```
Comment (jsomsanith-tlnd) [inline packages/config-react-webpack/config/utils/ts-config-exists.js:4]: This is to mutualize the same way to see if we use typescript betwee nthe app config and the umd lib config

File: packages/config-react-webpack/index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+module.exports = function getWebpackConfiguration(presetApi, option) {
+	if (option?.umd) {
+		const getUMDWebpackConfig = require('./config/webpack.config.umd');
+		return getUMDWebpackConfig(presetApi);
+	}
 
-module.exports = function getWebpackConfiguration(presetApi) {
+	const getCommonWebpackConfig = require('./config/webpack.config');
+	const getDevWebpackConfig = require('./config/webpack.config.dev');
+	const getProdWebpackConfig = require('./config/webpack.config.prod');
```
Comment (jsomsanith-tlnd) [inline packages/config-react-webpack/index.js:9]: Move the `require()` here to void importing them when we want to build an umd lib. The issue is that those configurations depends on @talend/react-components for the app loader, and e don't want to install that

File: packages/core/config/webpack.config.js [hunk 3]
```diff
@@ -18,16 +18,19 @@ module.exports = env => {
...
 	if (env && env.analyze) {
 		webpackConfigurations.push(require('./webpack.analyzer'));
 	}
 
+	webpackConfigurations = webpackConfigurations.map(conf => {
+		if (typeof conf === 'function') {
+			return conf(env);
+		}
+		return conf;
+	});
```
Comment (jsomsanith-tlnd) [inline packages/core/config/webpack.config.js:32]: Before, if the user specific conf was a function, we call it passing the env variables from webpack. But we want to do that with our internal conf too for the umd configuration

File: packages/core/doc/getting-started.md [hunk 4]
```diff
@@ -21,84 +24,91 @@ yarn add --dev @talend/scripts-preset-react-ng
...
 ```json
 {
-  "prepublish": "talend-scripts build:lib",
+	"prepare": "npm run build:lib && build:lib:umd && build:lib:umd:min",
 
-  "test": "talend-scripts test",
-  "test:watch": "talend-scripts test --watch",
+	"build:lib": "talend-scripts build:lib",
+	"build:lib:umd": "talend-scripts build:lib --env.umd=MyLibraryName",
+	"build:lib:umd:min": "talend-scripts build:lib --env.umd=MyLibraryName --env.optimize",
```
Comment (jsomsanith-tlnd) [inline packages/core/doc/getting-started.md:87]: Doc to build via babel + webpack umd + webpack umd minified

File: packages/preset-react-lib/index.js [hunk 5]
```diff
@@ -2,10 +2,12 @@ const getBabelConfigurationPath = require('@talend/scripts-config-babel');
...
 const getJestConfigurationPath = require('@talend/scripts-config-jest');
 const getTypescriptConfigurationPath = require('@talend/scripts-config-typescript');
+const getWebpackConfiguration = require('@talend/scripts-config-react-webpack');
 
 module.exports = {
 	getEslintConfigurationPath,
 	getBabelConfigurationPath,
 	getJestConfigurationPath,
 	getTypescriptConfigurationPath,
+	getWebpackConfiguration: presetApi => getWebpackConfiguration(presetApi, { umd: true }),
```
Comment (jsomsanith-tlnd) [inline packages/preset-react-lib/index.js:12]: preset react-lib need to provide a webpack config for UMD build

File: packages/config-react-webpack/config/webpack.config.js [hunk 6]
```diff
@@ -15,22 +15,19 @@ const ReactCMFWebpackPlugin = require('@talend/react-cmf-webpack-plugin');
 const TalendHTML = require('@talend/html-webpack-plugin');
 const icons = require.resolve('@talend/icons/dist/info');
 const AppLoader = require('@talend/react-components/lib/AppLoader/constant').default;
+const { getBabelConfig } = require('@talend/scripts-config-babel/babel-resolver');
+
+const LICENSE_BANNER = require('./licence');
```
Comment (frassinier) [inline packages/config-react-webpack/config/webpack.config.js:20]: or license yes :)

File: packages/config-react-webpack/config/webpack.config.umd.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+const { getBabelConfig } = require('@talend/scripts-config-babel/babel-resolver');
+
+const tsConfigExists = require('./utils/ts-config-exists');
+
+module.exports = () => {
+	const useTypescript = tsConfigExists();
+
+	return env => {
+		const name = env?.umd || 'Talend';
+		const optimize = !!env?.optimize;
```
Comment (frassinier) [inline packages/config-react-webpack/config/webpack.config.umd.js:11]: I've encountered some issue using node v12 with this

**Pull request merged**